src/DrawManager.sol

```diff
@@ -270,121 +309,138 @@ contract DrawManager {
       revert RewardRecipientIsZero();
     }
 
-    StartDrawAuction memory startDrawAuction = _lastStartDrawAuction;
+    StartDrawAuction memory startDrawAuction = getLastStartDrawAuction();
     
     if (startDrawAuction.drawId != prizePool.getDrawIdToAward()) {
       revert DrawHasFinalized();
     }
 
-    if (!rng.isRequestComplete(_lastStartDrawAuction.rngRequestId)) {
+    if (!rng.isRequestComplete(startDrawAuction.rngRequestId)) {
       revert RngRequestNotComplete();
     }
 
-    if (_hasAuctionExpired(startDrawAuction.startedAt)) {
+    if (_isAuctionExpired(startDrawAuction.closedAt)) {
       revert AuctionExpired();
     }
-
-    uint totalReserve = prizePool.reserve() + prizePool.pendingReserveContributions();
-    uint48 closesAt = prizePool.drawClosesAt(startDrawAuction.drawId);    
-    uint48 startDrawElapsedTime = _elapsedTimeSinceDrawClosed(startDrawAuction.startedAt, closesAt);
-    uint48 finishDrawElapsedTime = uint48(block.timestamp - startDrawAuction.startedAt);
-
-    (uint256[] memory rewards, uint256 remainingReserve) = computeRewards(startDrawElapsedTime, finishDrawElapsedTime, totalReserve);
-
+    
+    StartDrawAuction memory lastAuction = getLastStartDrawAuction();
+    uint256 availableRewards = _computeAvailableRewards();
+    (uint256[] memory startDrawRewards, UD2x18[] memory startDrawFractions) = _computeStartDrawRewards(prizePool.drawClosesAt(lastAuction.drawId), 0, availableRewards);
+    (uint256 _finishDrawReward, UD2x18 finishFraction) = _computeFinishDrawReward(lastAuction.closedAt, block.timestamp, availableRewards);
```

Comment:
Since the reward calculations all assume they have access to the same available rewards we can end up in a scenario where the start draw(s) are completed, but there is not enough liquidity to award them all. The `finishDraw` call would revert and the bots that completed the start draw(s) would not be rewarded.